components/ai_chat/core/common/mojom/ai_chat.mojom

```diff
@@ -195,6 +195,31 @@ struct LeoModelOptions {
 
 struct CustomModelOptions {
   string model_request_name;
+  /**
+  * context_size:
+  *   This value is provided by the user when adding/configuring
+  *   a custom model. If not provided, Brave will default to using
+  *   ai_chat::kCustomModelDefaultContextSize.
+  */
+  uint32? context_size;
+  /**
+  * max_page_content_length:
+  *   Instructs handler's of the model to truncate page contents
+  *   when they exceed this limit. This is measured in characters,
+  *   not tokens. The value is determined at runtime based on the
+  *   context size and ai_chat::kStandardCharactersPerToken.
+  */
+  uint32? max_page_content_length;
+  /**
+  * long_conversation_warning_character_limit:
+  *   Instructs handler's of the model as to the overall limits of
+  *   the conversation. If the conversation exceeds this limit, it
+  *   is likely that coherence and recall will be impacted. This is
+  *   measured in characters, not tokens. This value is also
+  *   determined at runtime based on the model's context size and
+  *   ai_chat::kStandardCharactersPerToken.
+  */
+  uint32? long_conversation_warning_character_limit;
```

Comment:
Do you think it's useful to even offer the users fill this in? How about we keep it a little simpler and automatically give the warning when we're some fixed number (e.g. 800) or percentage (e.g. 85%) away from the context limit? Since it doesn't affect functionality and isn't really a param. After all, the user is making this for themselves and not for other users of their own.

Reply:
Users don't ever see these values anywhere; they exist to give custom models symmetry with leo models, making client-side JavaScript a bit cleaner. The values are populated once the models are loaded, and again if/when a model is changed. We could remove these, but I would suggest keeping them as long as we have the same properties on Leo models.